sound/soc/sof/topology.c

```diff
@@ -1749,7 +1749,7 @@ static int sof_link_load(struct snd_soc_component *scomp, int index,
 		memcpy(&dai->dai_config, &config,
 		       sizeof(struct sof_ipc_dai_config));
 
-	return 0;
+	return ret;
```

Comment:
wondering if we should also return in the default case line 1743/44, it makes no sense to provide an error message and then still call snd_sof_find_dai.